### PR TITLE
Add Mavis Market to nft.trades

### PR DESCRIPTION
**Pull request by peterrliem**
## Thank you for contributing to Spellbook 🪄
Please open the PR in **draft** and mark as ready when you want to request a review. 

### Description:

[...]

---
quick links for more information:
- [README.md](https://github.com/duneanalytics/spellbook/blob/main/README.md)
- [spellbook docs](https://github.com/duneanalytics/spellbook/tree/main/docs)
- [CONTRIBUTING.md](https://github.com/duneanalytics/spellbook/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Murshadddd** [comment]
#7234

File: dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  block_time,
+  cast(date_trunc('day', block_time) as date) as block_date,
+  cast(date_trunc('month', block_time) as date) as block_month,
+  block_number,
+  nft_contract_address,
+  nft_token_id,
+  quantity as nft_amount,
+  seller,
+  buyer,
+  '' as trade_category,
```
Comment (0xRobin) [inline dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql:69]: This should be 'Buy' or 'Sell'.
If it's not possible to determine the trade direction, feel free to just set 'Buy' as the default.

File: dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  '' as trade_type,
```
Comment (0xRobin) [inline dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql:70]: This should be 'secondary'

File: dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  evt_tx_to as tx_to,
+  evt_index,
+  contract_address,
+  COALESCE(FROM_HEX(json_extract_scalar(json_parse(cast(concat('[', array_join(receivedAllocs, ','), ']') as varchar)),'$[5].recipient')),FROM_HEX(json_extract_scalar(json_parse(json_extract_scalar("order", '$.info')), '$.maker'))) as seller,
+  FROM_HEX(json_extract_scalar("order", '$.recipient')) as buyer,
+  json_extract_scalar(json_parse(json_extract_scalar("order", '$.info')), '$.kind') as kind,
+  cast(json_extract_scalar(replace(json_extract_scalar(json_parse(json_extract_scalar("order", '$.info')),'$.assets[0]'),'\\', ''),'$.erc') as double) as erc,
+  FROM_HEX(json_extract_scalar(replace(json_extract_scalar(json_parse(json_extract_scalar("order", '$.info')),'$.assets[0]'),'\\', ''),'$.addr')) as nft_contract_address,
+  cast(json_extract_scalar(replace(json_extract_scalar(json_parse(json_extract_scalar("order", '$.info')),'$.assets[0]'),'\\', ''),'$.id') as double) as nft_token_id,
+  cast(json_extract_scalar(replace(json_extract_scalar(json_parse(json_extract_scalar("order", '$.info')),'$.assets[0]'),'\\', ''),'$.quantity') as double) as quantity,
```
Comment (0xRobin) [inline dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql:29]: This seems to return 0 in quite some cases, it should always be 1 or greater

File: dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  seller,
+  buyer,
+  '' as trade_category,
+  '' as trade_type,
+  price_raw,
+  currency_address as currency_contract,
+  contract_address as project_contract_address,
+  tx_hash,
+  platform_address as platform_fee_address,
+  platform_fee_amount_raw,
```
Comment (0xRobin) [inline dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql:76]: This always returns `200` currently which seems to be the ratio and not the actual raw value?

File: dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  creator_royalty_address as royalty_fee_address,
+  creator_royalty_fee_amount_raw as royalty_fee_amount_raw,
+  -- axie_fee_amount_raw,
+  -- ronin_treasury_fee_amount_raw,
```
Comment (0xRobin) [inline dbt_subprojects/nft/models/_sector/trades/chains/ronin/platforms/mavis_ronin_base_trades.sql:80]: if you want, you can keep these in this model. 
They won't make it into `nft.trades` but at least they'll be available in the project specific model.

**Review by 0xRobin — CHANGES_REQUESTED** [review]
Left some questions/comments around data QA.
It would also be nice to add a seed test to the model. 
You can see how that is done for collectionswap here:
seed: https://github.com/duneanalytics/spellbook/blob/main/dbt_subprojects/nft/seeds/trades/base_trades/collectionswap_ethereum_base_trades_seed.csv
schema: https://github.com/duneanalytics/spellbook/blob/6a2d09f5025520e2b1db53453c73e75a7e303ad2/dbt_subprojects/nft/seeds/trades/base_trades/_schema.yml#L196-L206
test: https://github.com/duneanalytics/spellbook/blob/6a2d09f5025520e2b1db53453c73e75a7e303ad2/dbt_subprojects/nft/models/_sector/trades/chains/ethereum/platforms/_schema.yml#L573-L586

**Review by 0xRobin — APPROVED** [review]

**Pull request merged**